src/globals.c

```diff
@@ -543,3 +545,10 @@ void free_globals(void)
 		bundle_to_add = NULL;
 	}
 }
+
+void save_cmd(int argc, char **argv)
+{
+	for (int i = 0; i < argc; i++) {
+		string_or_die(&swupd_cmd, "%s %s", swupd_cmd, argv[i]);
```

Comment:
This will just overwrite previous flags from argv as it loops if there is more than 1 flag, and leak all the previous ones since it sets the pointer *swupd_cmd to a new one with string_or_die(). So when this loop finishes, you only have saved argv[n] where n is the last argument passed, and none of the others. Need to append/concat to the &swupd_cmd as you're iterating through argv insead. You can also create a swupd_cmds list of size argc-1 and do something like:
`string_or_die(&swupd_cmd[i]...` to save each flag into the list.